ts/javascript/vite.ts

```diff
@@ -0,0 +1,26 @@
+import { Executable } from "../executable.ts";
+import { Package } from "../package.ts";
+import { Plugin } from "../project.ts";
+
+export const plugin: Plugin<
+  { build: Package; dev: Executable; preview: Executable },
+  { node_modules: Package }
+> = (base) => {
+  if (base.defaultEnvironment == null) {
+    throw new Error(
+      `'vite.addBuilder' can only be added to projects with a default environment`,
+    );
+  }
+  return {
+    build: base.defaultEnvironment.build`
+      export PATH=${base.node_modules}/bin:$PATH
+      vite build --outDir $out
+    `,
+    dev: base.defaultEnvironment.shell`
+      export PATH=${base.node_modules}/bin:$PATH
+      vite`,
+    preview: base.defaultEnvironment.shell`
+      export PATH=${base.node_modules}/bin:$PATH
```

Comment:
Maybe this should be set by the environment, but up to you.

Reply:
I think it should be, but not in this PR, since I think there's some other things that we want to change then. For example instead of having the initializer generate `.addExecutable("npm run foo")` we could have it put in the actual script snippet from the `package.json` file.